crates/ccdi-server/src/filter/sample.rs

```diff
@@ -36,82 +38,116 @@ impl FilterMetadataField<Sample, FilterSampleParams> for Vec<Sample> {
-                    "age_at_diagnosis" => sample
-                        .metadata()
-                        .and_then(|metadata| metadata.age_at_diagnosis())
-                        .map(|age_at_diagnosis| vec![age_at_diagnosis.to_string()]),
-                    "age_at_collection" => sample
-                        .metadata()
-                        .and_then(|metadata| metadata.age_at_collection())
-                        .map(|age_at_collection| vec![age_at_collection.to_string()]),
-                    "tumor_tissue_morphology" => sample
-                        .metadata()
-                        .and_then(|metadata| metadata.tumor_tissue_morphology())
-                        .map(|tumor_tissue_morphology| vec![tumor_tissue_morphology.to_string()]),
-                    "depositions" => sample
-                        .metadata()
-                        .and_then(|metadata| metadata.common().depositions())
-                        .map(|deposition| {
-                            deposition
-                                .iter()
-                                .cloned()
-                                .map(|accession| match accession {
-                                    Accession::dbGaP(accession) => accession.to_string(),
-                                })
-                                .collect::<Vec<String>>()
-                        }),
-                    _ => unreachable!("unhandled sample metadata field: {field}"),
-                };
+                // Search field matches by substring rather than exact match.
+                if field.as_str() == "search" {
+                    if let Some(metadata) = sample.metadata() {
+                        if let Some(diagnosis) = metadata.diagnosis() {
+                            // Only return the entry if the query is a substring
+                            // of the diagnosis, ignoring case.
+                            // Matching on to_lowercase is an approximation and will not cover
+                            // all unicode characters.
+                            let diagnosis_lower = diagnosis.to_string().to_lowercase();
+                            let query_lower = query.to_string().to_lowercase();
+                            return diagnosis_lower.to_string().contains(&query_lower);
+                        }
 
-                match values {
-                    Some(values) => values.into_iter().any(|s| s.eq(query)),
-                    // Samples with no values for this field are automatically
-                    // filtered as described in the rules for filtering.
-                    None => false,
+                        // If the metadata doesn't have a diagnosis, the entry
+                        // should not be included.
+                        return false;
+                    }
+
+                    // If no metadata is included, this entry should not be
+                    // included.
+                    false
```

Comment:
I assume you meant `return false`?

Reply:
@grugna Yes, I would assume that too! But when I changed it to `return false;` just now I got an error  from `cargo clippy -- -D warnings` linking to the [Clippy Lints: needless return](https://rust-lang.github.io/rust-clippy/master/index.html#needless_return) page.

So, I will leave it as `false` so that this code is more "rusty". Thanks for checking.